src/flashbar/__tests__/persistence-unit.test.tsx

```diff
@@ -0,0 +1,200 @@
+
+      const { container } = render(<Flashbar items={items} stackItems={true} />);
+      const wrapper = createWrapper(container).findFlashbar()!;
+
+      await waitFor(() => {
+        expect(wrapper.findItems()[0].findContent()!.getElement()).toHaveTextContent('Item 1');
+      });
+    });
+
+    it('updates notification bar count based on visible items', async () => {
+      mockRetrieveFlashbarDismiss.mockResolvedValue(true);
+      const items = createItems([true, true, false]);
+
+      const { container } = render(<Flashbar items={items} stackItems={true} />);
+      const wrapper = createWrapper(container).findFlashbar()!;
+
+      await waitFor(() => {
+        expect(wrapper.findItems()).toHaveLength(1);
+        expect(wrapper.findItems()[0].findContent()!.getElement()).toHaveTextContent('Item 2');
+      });
+    });
+  });
+
+  describe('Dismiss persistence', () => {
+    it('calls persistFlashbarDismiss and onDismiss when dismissing persistent item', async () => {
+      mockRetrieveFlashbarDismiss.mockResolvedValue(false);
+      const onDismissMocks = [jest.fn(), jest.fn(), jest.fn()];
+      const items = createItems([true, true, false], onDismissMocks);
+
+      const { container } = render(<Flashbar items={items} />);
+      const wrapper = createWrapper(container).findFlashbar()!;
+
+      await waitFor(() => {
+        expect(wrapper.findItems()).toHaveLength(3);
+      });
+
+      const item = wrapper.findItems()[0];
+      item.findDismissButton()!.click();
+
+      expect(mockPersistFlashbarDismiss).toHaveBeenCalledWith({ uniqueKey: 'key-0' });
+      expect(onDismissMocks[0]).toHaveBeenCalled();
+    });
+
+    it('does not call persistFlashbarDismiss for non-persistent items but calls onDismiss', async () => {
+      const onDismissMocks = [jest.fn(), jest.fn(), jest.fn()];
+      const items = createItems([true, true, false], onDismissMocks);
+
+      const { container } = render(<Flashbar items={items} />);
+      const wrapper = createWrapper(container).findFlashbar()!;
+
+      await waitFor(() => {
+        expect(wrapper.findItems()).toHaveLength(3);
+      });
+
+      const item = wrapper.findItems()[2];
+      item.findDismissButton()!.click();
+
+      expect(mockPersistFlashbarDismiss).not.toHaveBeenCalled();
+      expect(onDismissMocks[2]).toHaveBeenCalled();
+    });
+  });
+
+  describe('Persistence caching', () => {
+    it('does not re-check persistence for same items', async () => {
+      mockRetrieveFlashbarDismiss.mockResolvedValue(false);
+      const items = createItems([true]);
+
+      const { rerender } = render(<Flashbar items={items} />);
+
+      await waitFor(() => {
+        expect(mockRetrieveFlashbarDismiss).toHaveBeenCalledTimes(1);
+      });
+
+      rerender(<Flashbar items={items} />);
+      await waitFor(() => {
+        expect(mockRetrieveFlashbarDismiss).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    it('checks persistence for new items only', async () => {
```

Comment:
⛳  This misses one more use-case

1. Start with an empty array
2. Add one item, check that `mockRetrieveFlashbarDismiss` was called with that item id
3. Add one more item, check that `mockRetrieveFlashbarDismiss` was called with only the 2nd item id

My prediction - this test will fail and there is a bug in the code

Reply:
Added the test `checks persistence only for newly added items when start with empty array`